### DISCUSSION TITLE
Is there a prettier parser for AWS SES email templates?

### DISCUSSION THREAD

**Discussion by lexicalunit** [prettier/prettier#14943] [Q&A]
This is the handlebars-esque language that AWS uses for SES email templates:
```html
<html>
<body>
It looks {{variable}} like this {{#if whatever}} blah blah {{/if}}.
</body>
</html>
```

The default parser for `html` files will typically format code in these files like this:

```html
      <p>Due to the following error:</p>
      {{/if}} {{#if error_description}}
      <p>
        <b>{{error_description}}</b>
      </p>
      {{/if}}
```

It would be nicer if it understood the `{{}}` parts and didn't do thing like `{{/if}} {{#if error_description}}`. Sometimes it will even break them up like this:

```html
blah blah blah blah blah blah blah blah blah {{
if something}}
```

Which is bad.

I tried using `--parser glimmer` but that's not exactly right either. It doesn't understand some things that AWS SES template do allow, like changing context. For example the glimmer parser doesn't know how to handle the `../` part in something like this:

```html
{{#if ../data.foo}}
```

So what parser should I use?

**Comment by fisker** [comment]
There is no builtin parser for it. You can search/develop a plugin.
